widgetssdk/src/main/java/com/glia/widgets/engagement/completion/EngagementCompletionController.kt

```diff
@@ -37,20 +35,18 @@ internal class EngagementCompletionController(
 
     private fun handleEngagementState(state: State) {
         when (state) {
-            State.FinishedCallVisualizer, State.FinishedOmniCore -> {
-                releaseResourcesUseCase()
-                _state.onNext(EngagementCompletionState.QueuingOrEngagementEnded)
+            is State.EngagementEnded -> {
+                if (state.onEnd != Engagement.ActionOnEnd.RETAIN) releaseResourcesUseCase()
+                _state.onNext(EngagementCompletionState.EngagementEnded(state.isEndedByVisitor, state.onEnd))
```

Comment:
As I already mentioned, ideally, it might be better to use identical logical entity everywhere - either `isEndedByVisitor` or `isEndedByOperator`, but not both.